ground/src/main/java/com/google/android/ground/persistence/local/room/stores/RoomLocationOfInterestStore.kt

```diff
@@ -125,22 +126,25 @@ class RoomLocationOfInterestStore @Inject internal constructor() : LocalLocation
     }
   }
 
-  override fun getLocationOfInterestMutationsByLocationOfInterestIdOnceAndStream(
+  override fun getMutationsFlow(
     locationOfInterestId: String,
     vararg allowedStates: MutationEntitySyncStatus
-  ): Flowable<List<LocationOfInterestMutation>> =
-    locationOfInterestMutationDao
-      .findByLocationOfInterestIdOnceAndStream(locationOfInterestId, *allowedStates)
-      .map { list: List<LocationOfInterestMutationEntity> -> list.map { it.toModelObject() } }
+  ): Flow<List<LocationOfInterestMutation>> =
+    locationOfInterestMutationDao.getMutationsFlow(locationOfInterestId, *allowedStates).map {
+      mutations ->
+      mutations.map { it.toModelObject() }
+    }
 
-  override fun getAllMutationsAndStream(): Flowable<List<LocationOfInterestMutationEntity>> =
-    locationOfInterestMutationDao.loadAllOnceAndStream()
+  override fun getAllSurveyMutations(survey: Survey): Flow<List<LocationOfInterestMutation>> =
```

Comment:
Nit: Let's expose to min required args here, ie the survey ID. Then this method would be scoped at `getMutationsBySurveyId`.

Reply:
One downside to this is that all callers will need to perform conversion to model objects themselves. Right now we can define this as:

```kotlin
override fun getMutationsBySurveyId(survey: Survey): Flow<List<SubmissionMutation>> =
    submissionMutationDao.getAllMutationsFlow().map { mutations ->
      mutations.filter { it.surveyId == survey.id }.map { it.toModelObject(survey) }
    }
```

And callers don't need to know or care about performing the conversion to a model object. Contrarily, if we define this over IDs only, callers will need to do the conversion. This might be totally cool, since we'd still confine these conversions to the repository level, presumably, but we'd have more than one repository doing the same conversion just to carry across the Survey input.

I'm fine with either approach, just thought I'd flag it to see what you think.

Reply:
Ah sorry I missed the call to `it.toModelObject(survey)`, please disregard!